iml-system-test-utils/src/docker.rs

```diff
@@ -0,0 +1,223 @@
+use crate::{iml, CheckedStatus, iml::IML_DOCKER_PATH};
+use iml_systemd;
+use iml_wire_types::Branding;
+use std::{io, thread, time, str};
+use tokio::{
+    fs::{canonicalize, File},
+    io::AsyncWriteExt,
+    process::Command,
+};
+
+pub struct DockerSetup {
+  pub use_stratagem: bool,
+  pub branding: Branding,
+}
+
+pub async fn docker() -> Result<Command, io::Error> {
+    let mut x = Command::new("docker");
+
+    let path = canonicalize(IML_DOCKER_PATH).await?;
+
+    x.current_dir(path);
+
+    Ok(x)
+}
+
+pub async fn deploy_iml_stack() -> Result<(), io::Error> {
+    iml_systemd::start_unit_and_wait("iml-docker.service".into(), 400).await?;
+
+    Ok(())
+}
+
+pub async fn get_docker_service_count() -> Result<u16, io::Error> {
+  let mut x = docker().await?;
+
+  let services = x.arg("service").arg("ls").output().await?;
+
+  let output = str::from_utf8(&services.stdout).expect("Couldn't read docker service list.");
+  let cnt: u16 = output.lines().count() as u16 - 1; // subtract the header
+
+  Ok(cnt)
+} 
+
+pub async fn remove_iml_stack() -> Result<(), io::Error> {
```

Comment:
```suggestion
pub async fn remove_iml_stack() -> Result<(), iml_systemd::SystemdError> {
```

This will compile, and while not technically the best, it's still better than downsampling to io::Error from SystemdError.